### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by hajimefu**
A few patches to address Coverity defects.

### CODE REVIEW

**Comment by hajimefu** [comment]
test:intel:ch4/ofi/sockets,psm2/gnu/all/all/debug,default

**Comment by hajimefu** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by hajimefu** [comment]
test:intel:ch4/ofi/sockets,psm2/gnu/all/all/debug

File: src/mpid/ch4/src/ch4_globals.c [hunk 1]
```diff
@@ -152,21 +152,27 @@ int MPIDI_check_for_failed_procs(void)
...
                          len, &vallen);
         MPIR_ERR_CHKANDJUMP(pmi_errno, mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get");
+        MPL_free(kvsname);
     }
 #else
     pmi_errno = PMI_KVS_Get_value_length_max(&len);
     MPIR_ERR_CHKANDJUMP(pmi_errno, mpi_errno, MPI_ERR_OTHER, "**pmi_kvs_get_value_length_max");
+    kvsname = MPL_malloc(len, MPL_MEM_OTHER);
+    MPIR_Assert(kvsname);
     pmi_errno = PMI_KVS_Get_my_name(kvsname, len);
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4_globals.c]: So it seems that ch4 stores the kvsname value in the strangely named `MPIDI_CH4_Global.jobid`. We should use that and remove all the extra PMI calls here for now.

**Comment by hajimefu** [comment]
Thanks, update the `kvsname` patch, and fixed a commit message of the second patch.

**Comment by raffenet** [comment]
OK, let's make sure didn't break anything. I'm digging a little more into the errhan patch and coverity issues to understand it better in the meantime.
test:jenkins/ch4/ofi

**Comment by hajimefu** [comment]
Sorry, just realized that I did not remove a redundant PMI call for getting the kvsname. Just fixed it.
test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
OK, I think the errhan patch is worth a try. I removed some extra parens and will retest one more time.
test:jenkins/ch4/most

**Comment by raffenet** [comment]
Rebased and fixed conflict.
test:jenkins/ch4/most

**Comment by hajimefu** [comment]
No idea about the ucx failure, but I'm almost certain that they are not caused by this PR.

**Comment by raffenet** [comment]
Rebased one more time. The UCX problems should go away this time.
test:jenkins/ch4/ucx

**Comment by raffenet** [comment]
test:jenkins/ch4/ucx

**Comment by raffenet** [comment]
OK, I'm confidant these failures are not the fault of this PR. I am going to merge.

**Pull request merged**